src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/MetricsEventSource.cs

```diff
@@ -227,6 +239,12 @@ public CommandHandler(MetricsEventSource parent)
 
             public MetricsEventSource Parent { get; private set;}
 
+            public bool IsSharedSession(string commandSessionId)
+            {
+                // commandSessionId may be null if it's the disable command
+                return _sessionId == SharedSessionId && (string.IsNullOrEmpty(commandSessionId) || commandSessionId == SharedSessionId);
```

Comment:
Are session id's generally case sensitive? Also consider an explcit string.Equals instead of ==